src/lib/journey/journey.svelte

```diff
@@ -37,11 +43,19 @@
   {:else if $journeyStore.step.type === StepType.Step}
     <svelte:component
       this={mapStepToStage($journeyStore.step)}
-      {displayIcon}
-      failureMessage={$journeyStore.error?.message}
       bind:formEl
-      loading={$journeyStore.loading}
-      {submitForm}
+      form={{
+        icon: displayIcon,
+        message: $journeyStore.error?.message || '',
+        status: $journeyStore.error?.code ? 'error' : 'ok',
+        submit: submitForm,
+      }}
+      journey={{
+        loading: $journeyStore.loading,
+        pop: journeyStore.pop,
+        push: journeyStore.push,
+        stack,
+      }}
```

Comment:
I wanted to take this opportunity to start cleaning up and organizing the interfaces between the components. I think we can make it even better, but I needed to draw a line somewhere :)